clippy_lints/src/functions/impl_trait_in_params.rs

```diff
@@ -0,0 +1,88 @@
+    {
+        if let FnKind::ItemFn(ident, generics, _) = kind {
+            for param in generics.params {
+                if param.is_impl_trait()
+                    && !param.name.ident().as_str().contains('<')
+                    && !param.name.ident().as_str().contains('(')
+                {
+                    // No generics with nested generics, and no generics like FnMut(x)
+                    span_lint_and_then(
+                        cx,
+                        IMPL_TRAIT_IN_PARAMS,
+                        param.span,
+                        &format!("'{}' in the function's parameters", param.name.ident().as_str()),
+                        |diag| {
+                            let next_letter = next_valid_letter(generics);
+                            if let Some(gen_span) = generics.span_for_param_suggestion() {
+                                diag.span_suggestion_with_style(
+                                    gen_span,
+                                    format!(
+                                        "create a generic type here and replace that `{}` with `{}`",
+                                        param.name.ident().as_str(),
+                                        next_letter
+                                    ),
+                                    ", T: Trait",
+                                    rustc_errors::Applicability::MaybeIncorrect,
+                                    rustc_errors::SuggestionStyle::ShowAlways,
+                                );
+                            } else {
+                                // multispan.push_span_label(param.span, format!("Replace this with `{}`",
+                                // next_letter));
+
+                                diag.span_suggestion_with_style(
+                                    Span::new(
+                                        body.params[0].span.lo() - rustc_span::BytePos(1),
+                                        ident.span.hi(),
+                                        ident.span.ctxt(),
+                                        ident.span.parent(),
+                                    ),
+                                    format!(
+                                        "create a generic type here and replace that '{}' with `{}`",
+                                        param.name.ident().as_str(),
+                                        next_letter
+                                    ),
+                                    "<T: Trait>",
+                                    rustc_errors::Applicability::MaybeIncorrect,
+                                    rustc_errors::SuggestionStyle::ShowAlways,
+                                );
+                            }
+                        },
+                    );
+                }
+            }
+        }
+    }
+}
+
+fn next_valid_letter(generics: &Generics<'_>) -> char {
```

Comment:
ping @blyxyas. Are you still working on this?

Reply:
Yes I am still working on it. I thought the lint didn't need it because "Just using a placeholder is also a reasonable option" so I didn't change anything. Sorry for not seeing the ping before. I'll implement this right now.